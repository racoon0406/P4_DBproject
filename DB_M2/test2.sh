#!/bin/bash
# test2: tests additional put and get requests with versioning and edge cases
#        also test load balancing and failure detection behavior (PING/PONG)

# put values into database
./send.py put 10 100
./send.py put 10 101
./send.py put 10 102
./send.py put 10 103
./send.py put 10 104
./send.py put 10 105
./send.py put 11 110
./send.py put 11 111
./send.py put 11 112
./send.py put 12 120
./send.py put 12 121
./send.py put 12 122

./send.py put 512 51200
./send.py put 512 51201
./send.py put 512 51202
./send.py put 513 51300
./send.py put 513 51301
./send.py put 513 51302

# put key that already has 6 versions in database
./send.py put 10 106
./send.py put 10 107

# get value that exists
./send.py get 10 0
./send.py get 10 1
./send.py get 10 2
./send.py get 10 3
./send.py get 10 4
./send.py get 10 5
./send.py get 11 0
./send.py get 11 1
./send.py get 11 2
./send.py get 12 0
./send.py get 12 1
./send.py get 12 2
./send.py get 512 0
./send.py get 512 1
./send.py get 512 2
./send.py get 513 0
./send.py get 513 1
./send.py get 513 2

# get version out of bound or does not exist
./send.py get 10 6
./send.py get 10 10

# get version does not exist
./send.py get 11 3
./send.py get 12 4
./send.py get 512 5

# get value does not exist
./send.py get 13 0
./send.py get 14 0
./send.py get 25 0
./send.py get 514 0