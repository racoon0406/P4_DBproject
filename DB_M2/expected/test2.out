sniffing on eth0
try to put(key= 10, value= 100) to switch 3
try to put(key= 10, value= 100) to switch 1
try to put(key= 10, value= 101) to switch 3
try to put(key= 10, value= 101) to switch 1
try to put(key= 10, value= 102) to switch 3
try to put(key= 10, value= 102) to switch 1
try to put(key= 10, value= 103) to switch 1
try to put(key= 10, value= 103) to switch 3
try to put(key= 10, value= 104) to switch 3
try to put(key= 10, value= 104) to switch 1
try to put(key= 10, value= 105) to switch 3
try to put(key= 10, value= 105) to switch 1
try to put(key= 11, value= 110) to switch 1
try to put(key= 11, value= 110) to switch 3
try to put(key= 11, value= 111) to switch 1
try to put(key= 11, value= 111) to switch 3
try to put(key= 11, value= 112) to switch 1
try to put(key= 11, value= 112) to switch 3
try to put(key= 12, value= 120) to switch 3
try to put(key= 12, value= 120) to switch 1
try to put(key= 12, value= 121) to switch 1
try to put(key= 12, value= 121) to switch 3
try to put(key= 12, value= 122) to switch 3
try to put(key= 12, value= 122) to switch 1
try to put(key= 512, value= 51200) to switch 3
try to put(key= 512, value= 51200) to switch 1
try to put(key= 512, value= 51201) to switch 3
try to put(key= 512, value= 51201) to switch 1
try to put(key= 512, value= 51202) to switch 1
try to put(key= 512, value= 51202) to switch 3
try to put(key= 513, value= 51300) to switch 3
try to put(key= 513, value= 51300) to switch 2
try to put(key= 513, value= 51301) to switch 3
try to put(key= 513, value= 51301) to switch 2
try to put(key= 513, value= 51302) to switch 3
try to put(key= 513, value= 51302) to switch 2
try to put(key= 10, value= 106) to switch 3
try to put(key= 10, value= 106) to switch 1
try to put(key= 10, value= 107) to switch 1
try to put(key= 10, value= 107) to switch 3
get value from switch 1
100
get value from switch 1
101
get value from switch 1
102
get value from switch 1
103
get value from switch 1
104
get value from switch 1
105
get value from switch 1
110
received a PONG from switch 1
received a PONG from switch 2
get value from switch 1
111
get value from switch 1
112
get value from switch 1
120
get value from switch 1
121
get value from switch 1
122
get value from switch 1
51200
get value from switch 1
51201
get value from switch 1
51202
get value from switch 2
51300
get value from switch 2
51301
get value from switch 2
51302
get value from switch 1
NULL
get value from switch 1
NULL
get value from switch 1
NULL
get value from switch 1
NULL
get value from switch 1
NULL
received a PONG from switch 1
received a PONG from switch 2
get value from switch 1
NULL
get value from switch 2
NULL
